pull_request_template.md

```diff
@@ -1,4 +1,124 @@
+
+    + <details> <summary>Documentation fix</summary>
+
+        - The modification was to [developer/user] documentation
+        - The change was a [correction/deletion/addition]
+        - The reason for the change was [...]
+        - The documentation component that changed was [...]
+
+      </details>
+
+    + <details> <summary>Bug fix</summary>
+
+        - [ ] Details about the bug are located in the above bug report
+        - [ ] Regression test for the issue was added
+        - [ ] Fix has been verified
+
+      </details>
+
+    + <details> <summary>New methods added to a C++ class</summary>
+
+        - [ ] Unit tests added for new class methods except: non-test
+              constructors, private methods or classes that provide a
+              last-level-abstraction of an external dependency
+              (e.g. SDBus or NVMLDevicePool)
+        - [ ] Non-exempt methods with logic branches have multiple
+              tests
+        - [ ] Mock objects derived from class have methods added
+        - [ ] New public methods have doxygen comments describing
+              behavior, inputs, outputs and exceptions raised
+
+      </details>
+
+    + <details> <summary>New methods added to Python module</summary>
+
+        - [ ] Unit test added for new public methods
+        - [ ] Unit tests do not refer to private methods or members
+        - [ ] New public methods have __doc__ string comments describing
+              behavior, inputs, outputs and exceptions raised
+
+      </details>
+
+    + <details> <summary>New external dependency</summary>
+
+        - [ ] Use of interface exposed by external dependency is
+              compatible with the BSD 3 clause license
+        - [ ] Shared object dependencies have been added to the build
+              system in this or a prior PR
+        - [ ] Dependency has one or more last-level-abstraction classes
+              if used in C++
+        - [ ] Dependency may have last-level-abstraction when used in
+              Python if this enables testing
+        - [ ] All existing features continue to work when dependency is
+              not available
+
+      </details>
+
+    + <details> <summary>A new C++ class added</summary>
+
+        - [ ] Pure virtual interface class has been added with factory method
+        - [ ] Unit test constructor provides dependency injection
+        - [ ] Unit tests added for all class methods except: non-test
+              constructors, private methods or classes that provide a
+              last-level-abstraction of an external dependency
+              (e.g. SDBus or NVMLDevicePool)
+        - [ ] Non-exempt methods with logic branches have multiple
+              tests
+        - [ ] All dependency classes have already been merged
+        - [ ] All public methods have doxygen comments describing
+              behavior, inputs, outputs and exceptions raised
+        - [ ] Implementation file has `#include "config.h"`
```

Comment:
This might be something we could automate, either by guaranteeing at compile time or by verifying at test time. Opened #1890 as a tech-debt issue to automate this.